### ISSUE TITLE
Add benchmarks

### ISSUE THREAD

**Issue by zrzka** [crossterm-rs/crossterm#334]
Still unclear if we will use `anes-rs` parser or when, but it's not important now, because the parser isn't finished (still WIP).

I made some changes in the `zrzka/performance` branch based on the manual profiling of the code with Instruments, because of the https://github.com/gyscos/cursive/pull/406 PR.

But we should provide benchmarks (especially for events parsing), so we can check that there's no regression after every change we make in the `event` module. `crossterm` is used as a backend for the `cursive` crate with 36 dependent crates and any regression can influence all of them. Not talking about projects (applications) that are not published to crates.io (not listed in the dependent crates).

**Comment by TimonPost** [comment]
Yes, we will be using that. Once it is ready we can do an 0.14.* release.
